File: objc/Builder/Builder/ComputerBuilder.m
//
//  ComputerBuilder.m
//  Builder
//
//  Created by mahbub on 9/14/17.
//  Copyright © 2017 KAZI MAHBUBUR RAHMAN. All rights reserved.
//

#import "ComputerBuilder.h"

@implementation ComputerBuilder

-(Computer*)getComputer {
    return _computer;
}

-(void)createComputer {
    _computer = [[Computer alloc] init];
}

@end

File: swift/Builder/Builder/ComputerBuilder.swift
import Foundation

protocol ComputerBuildProtocol {
    func buildCPU() -> Void
    func buildMotherboard() -> Void
    func buildRAM() -> Void
    func buildStorage() -> Void
    func buildGraphicsCard() -> Void
}

public class ComputerBuilder : ComputerBuildProtocol {
    var computer: Computer?
    
    func getComputer() -> Computer {
        return self.computer!
    }
    
    func createComputer() -> Void {
        self.computer = Computer()
    }
    
    func buildCPU() {}
    
    func buildMotherboard() {}
    
    func buildRAM() {}
    
    func buildStorage() {}
    
    func buildGraphicsCard() {}
}
